lib/irq.c

```diff
@@ -171,8 +171,13 @@ irqs_set_data_eventfd(vfu_ctx_t *vfu_ctx, struct vfio_irq_set *irq_set,
 
             vfu_ctx->irqs->efds[i] = -1;
         }
-        if (*d32 >= 0) {
-            vfu_ctx->irqs->efds[i] = *d32;
+        if (data[j] >= 0) { /* TODO IIUC this will always be >= 0? */
+            vfu_ctx->irqs->efds[i] = data[j];
```

Comment:
you can now do:

vfu_ctx->irqs->efds[i] = consume_fd(data, irq_set->count, j);

As for the TODO: why not add an assert in consume_fd() that fd is >= 0?

Reply:
That was my intention, I forgot to do that.

Also, replaced the TODO with an assert.